### PR TITLE
Adding storage form

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

> Storage Expandable section for disk source customization

## 🎥 Demo
Screenshot from 2022-03-09 16-34-26

Screenshot from 2022-03-09 16-34-34

Screenshot from 2022-03-09 16-34-49

### CODE REVIEW

**Comment by upalatucci** [comment]
@glekner @yaacov @metalice @avivtur @pcbailey @hstastna @vojtechszocs Can you please review this pr? :grin:

File: src/views/catalog/customize/components/PersistentVolumeClaimSelect/PersistentVolumeSelectProject.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  return (
+    <FormGroup
+      label={t('Persistent Volume Claim project')}
+      fieldId={fieldId}
+      id={fieldId}
+      isRequired
+      className="pvc-selection-formgroup"
+    >
+      <Select
```
Comment (glekner) [inline src/views/catalog/customize/components/PersistentVolumeClaimSelect/PersistentVolumeSelectProject.tsx:47]: add `maxHeight={400}` to make the inner select scrollable.
Add it to all `Select` instances actually

File: src/views/catalog/customize/components/CustomizeSource/CustomizeSource.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+          label={t('Image URL')}
+          fieldId={`disk-source-required-${selectedSourceType}`}
+          isRequired
+          className="disk-source-form-group"
+        >
+          <TextInput
+            value={httpURL}
+            type="text"
+            onChange={setHTTPURL}
+            aria-label={t('Image URL')}
```
Comment (glekner) [inline src/views/catalog/customize/components/CustomizeSource/CustomizeSource.tsx:144]: we need to have better validation here. at least `startsWith('https://...)` cc @yaacov

File: src/views/catalog/customize/components/CustomizeSource/quantity.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import byteSize, { ByteSizeResult } from 'byte-size';
+
+const multipliers: Record<string, number> = {};
```
Comment (glekner) [inline src/views/catalog/customize/components/CustomizeSource/quantity.ts:3]: @avivtur didn't you implement this already in the disks shared files?
we need a share util for units, sizes

**Review by glekner — COMMENTED** [review]
looks good @upalatucci.
we need to think about shared util for units, sizes

File: src/views/catalog/customize/components/PersistentVolumeClaimSelect/PersistentVolumeSelectProject.tsx (hunk 1 above)
Comment (upalatucci) [reply src/views/catalog/customize/components/PersistentVolumeClaimSelect/PersistentVolumeSelectProject.tsx:47]: okay thanks

File: src/views/catalog/customize/components/CustomizeSource/CustomizeSource.tsx (hunk 2 above)
Comment (upalatucci) [reply src/views/catalog/customize/components/CustomizeSource/CustomizeSource.tsx:144]: Okay for me. Is there also for the container URL some simple validation that we can use?

Comment (yaacov) [reply src/views/catalog/customize/components/CustomizeSource/CustomizeSource.tsx:144]: the validation is "nice to have", the requirement is to have a text input.

lets start with something working, and then after we have it working add the validation, in the next step

File: src/views/catalog/customize/components/CustomizeSource/quantity.ts (hunk 3 above)
Comment (yaacov) [reply src/views/catalog/customize/components/CustomizeSource/quantity.ts:3]: we are currently in the phase of getting things done for each plugin extension,
it means that @avivtur may have his own version of the same functionlity,

once we have a working extension, we may go over it, choose the best implementation and move it to a shared directory

File: src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  const [requiredFields, optionalFields] = buildFields(template, [parameterForName], t);
+
+  const onDiskSourceChange = React.useCallback((newDiskSource) => {
+    setCustomDiskSource(newDiskSource);
+  }, []);
+
+  return (
+    <Form onSubmit={onSubmit}>
+      {requiredFields?.map((field) => (
```
Comment (yaacov) [inline src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx:39]: name override inputs are above strorage section
required fields are below the storage section

---
order of inputs is:

name
storage (disk or cd)
add drivers disk check box
required parameters
optional parameters

File: src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        <FieldGroup key={field.name} field={field} showError={error} />
+      ))}
+
+      <ExpandableCustomizeSourceSection
+        onChange={onDiskSourceChange}
```
Comment (yaacov) [inline src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx:44]: you can't have a different onChange for the different sections, 
it's always
a - pass parameters
b - overide name
c - overide storage
d - add drivers disk
in one pass
there is no meaning to doing on change without first processing the parameters

Comment (yaacov) [reply src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx:44]: to clearify
you should have one onChange method
this method should
a - cakk a method that takes a template name+namespace + parameters and return a processed templates
b - take the vm object
c - update vm name and namespace
d - update main disk
e - add installCD if needed
f - add driversCD if needed
return the vm json ( do with it what needed for example create the VM / store it in some config map as json string / store it in session storage ... )

File: src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx (hunk 4 above)
Comment (upalatucci) [reply src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx:39]: really? So strange

File: src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx (hunk 5 above)
Comment (upalatucci) [reply src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx:44]: okay, so the problem is that this process should not be in the onSubmit but in the onChange right?

File: src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx (hunk 4 above)
Comment (upalatucci) [reply src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx]: fixed now

**Pull request merged**